### ISSUE TITLE
Use `tempfile` module for temporary files

### ISSUE THREAD

**Issue by knikolla** [CCI-MOC/MOC-issues#279]
> I would like to see all of these items being downloaded into `/tmp` converted to use `tempfile.TemporaryFile` (or `tempfile.NamedTemporaryFile`, if necessary) so that they are automatically cleaned up, but I think that may be for a future PR (because it will require some minor re-architecting of the code).

_Originally posted by @larsks in https://github.com/CCI-MOC/openstack-billing-from-db/pull/37#discussion_r1480430075_

**Comment by QuanMPhm** [comment]
@knikolla @larsks From my understanding, the script downloads the Nova DB dump from S3, optionally decompresses it, then optionally converts it to be SQLite compatible, so up to 3 DB files may be created. All 3 DB files are stored in `/tmp/`. Do we want to remove all 3 after the billing script finishes?

In `invoicing`, all the files we fetch from S3 or elsewhere remains on the system. Keeping them there was quite useful for testing local. Let me know your thoughts.

**Comment by knikolla** [comment]
@QuanMPhm yes, we want to clean up what this script downloads after it finishes.

In `invoicing` it makes sense to keep them there (and AFAIK they're not in `/tmp/`.
